2958 add redetect functionality

Zeskbest opened the pull request.

Commit:
add redetect functionality (fsdk rc_3.9.0)

matemax commented on lunavl/sdk/errors/errors.py:
```diff
@@ -134,11 +134,11 @@ def fromSDKError(cls, sdkError: FSDKErrorResult) -> "ErrorInfo":
-        def getError(sdkError: FSDKErrorResult):
-            attrs = dir(sdkError)
+        def getError(sdkError_: FSDKErrorResult):
+            attrs = dir(sdkError_)
```
I think that FaceEngine  removed differents error. Now  FSDKErrorResult has an uniforf attributes for error

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -22,13 +22,39 @@ class ImageForDetection(NamedTuple):
+    """
+
+    image: VLImage
+    BBoxes: List[Rect[float]]
```
BBoxes - style

matemax replied:
Need support `Rect[int]`

Zeskbest replied:
done

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +339,30 @@ def detect(
-    def redetectOne(self):
+    def redetectOne(self, image: ImageForRedetection):
         """
         todo: wtf
```
comments

Zeskbest replied:
done

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +339,30 @@ def detect(
         return res
 
-    def redetectOne(self):
+    def redetectOne(self, image: ImageForRedetection):
```
I thin, that  we should support  following interface for usability:
 `ImageForRedetection`
 `VLImage` + `Rect`
 `VLImage`  + detection

matemax replied:
Also need @CoreExceptionWarp

Zeskbest replied:
done

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +339,30 @@ def detect(
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        return detectRes
+
+    def redetect(self, images: List[ImageForRedetection]):
```
Need @CoreExceptionWarp

Zeskbest replied:
done

matemax commented on lunavl/sdk/luna_faces.py:
```diff
@@ -324,6 +327,40 @@ def detect(self, images: List[Union[VLImage, ImageForDetection]], limit: int = 5
             )
         return res
 
+    def redetectOne(self, image: ImageForRedetections) -> List[VLFaceDetection]:
```
redetectOne return <= 1 Face

Zeskbest replied:
no! it returns a detection or raises an exception

matemax commented on lunavl/sdk/luna_faces.py:
```diff
@@ -324,6 +327,40 @@ def detect(self, images: List[Union[VLImage, ImageForDetection]], limit: int = 5
+            res.append(VLFaceDetection(redetection, redetection.img, self.estimatorsCollection))
+        return res
+
+    def redetect(self, images: List[ImageForRedetections]) -> List[List[VLFaceDetection]]:
```
Comments

Zeskbest replied:
done

matemax commented on lunavl/sdk/luna_faces.py:
```diff
@@ -324,6 +327,40 @@ def detect(self, images: List[Union[VLImage, ImageForDetection]], limit: int = 5
 
+    def redetectOne(self, image: ImageForRedetections) -> List[VLFaceDetection]:
+        """
+        Redetect faces on an image.
```
face

Zeskbest replied:
No. It really detects several faces.

matemax commented:
Pleaze, update docs and examples

Commit:
fix FaceEngine.Rect and FaceEngine.RectFloat processing;
add comments

Commit:
add variety for facedetector.redetectOne call
ImageForRedetection.BBox can be Rect[int] also
rename and use CoreExceptionWrap
add detect5Landmarks and detect68Landmarks flags to redetect functions
add more comments

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -22,13 +22,26 @@ class ImageForDetection(NamedTuple):
+    """
+
+    image: VLImage
+    BBox: Union[Rect[float], Rect[float]]
```
Rect

Zeskbest replied:
ok

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +326,80 @@ def detect(
+            self,
+            image: Union[ImageForRedetection, VLImage],
+            BBox: Optional[Rect] = None,
+            detection: Optional[DetectionFloat] = None,
```
need FaceDetection instead DetectionFloat

Zeskbest replied:
ok

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +326,80 @@ def detect(
         """
-        todo: wtf
-        Returns:
+        Redetect one image.
```
I all understood...

Zeskbest replied:
Redetect faces on an image with restricted area.

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +326,80 @@ def detect(
+            BBox: Optional[Rect] = None,
+            detection: Optional[DetectionFloat] = None,
+            detect5Landmarks: bool = True,
+            detect68Landmarks: bool = False,
```
For what this useless parameters

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +326,80 @@ def detect(
+    def redetectOne(
+            self,
+            image: Union[ImageForRedetection, VLImage],
+            BBox: Optional[Rect] = None,
```
BBox - style

Zeskbest replied:
bBox now

matemax commented on lunavl/sdk/faceengine/facedetector.py:
```diff
@@ -313,21 +326,80 @@ def detect(
+    @CoreExceptionWrap(LunaVLError.DetectFacesError)
+    def redetectOne(
+            self,
+            image: Union[ImageForRedetection, VLImage],
```
use https://docs.python.org/3/library/typing.html#typing.overload

Zeskbest replied:
ok

matemax commented on lunavl/sdk/luna_faces.py:
```diff
@@ -324,6 +327,55 @@ def detect(self, images: List[Union[VLImage, ImageForDetection]], limit: int = 5
             )
         return res
 
+    def redetectOne(self, image: VLImage, BBoxes: List[Rect[float]]) -> List[VLFaceDetection]:
```
+ image and VLFaceDetection

Zeskbest replied:
ok

Commit:
review fixes

pep8speaks commented:
Hello @Zeskbest! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:































There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2019-10-18 07:52:26 UTC

Commit:
sdkError changes his interface

Commit:
syntax fix
